src/PrestaShopBundle/Controller/Admin/Sell/Address/AddressController.php

```diff
@@ -0,0 +1,292 @@
+use PrestaShop\PrestaShop\Core\Domain\Address\Query\GetRequiredFieldsForAddress;
+use PrestaShop\PrestaShop\Core\Search\Filters\AddressFilters;
+use PrestaShopBundle\Controller\Admin\FrameworkBundleAdminController;
+use PrestaShopBundle\Form\Admin\Sell\Address\RequiredFieldsAddressType;
+use PrestaShopBundle\Security\Annotation\AdminSecurity;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Class manages "Sell > Customers > Addresses" page.
+ */
+class AddressController extends FrameworkBundleAdminController
+{
+    /**
+     * Show addresses listing page
+     *
+     * @AdminSecurity("is_granted('read', request.get('_legacy_controller'))")
+     *
+     * @param Request $request
+     * @param AddressFilters $filters
+     *
+     * @return Response
+     */
+    public function indexAction(Request $request, AddressFilters $filters): Response
+    {
+        $addressGridFactory = $this->get('prestashop.core.grid.grid_factory.address');
+        $addressGrid = $addressGridFactory->getGrid($filters);
+        $requiredFieldsForm = $this->getRequiredFieldsForm();
+
+        return $this->render('@PrestaShop/Admin/Sell/Address/index.html.twig', [
+            'help_link' => $this->generateSidebarLink($request->attributes->get('_legacy_controller')),
+            'addressGrid' => $this->presentGrid($addressGrid),
+            'enableSidebar' => true,
+            'layoutHeaderToolbarBtn' => $this->getAddressToolbarButtons(),
+            'requiredFieldsForm' => $requiredFieldsForm->createView(),
+        ]);
+    }
+
+    /**
+     * Process addresses required fields configuration form.
+     *
+     * @AdminSecurity(
+     *     "is_granted(['update', 'create', 'delete'], request.get('_legacy_controller'))",
+     *     redirectRoute="admin_addresses_index"
+     * )
+     *
+     * @param Request $request
+     *
+     * @return RedirectResponse
+     *
+     * @throws Exception
+     */
+    public function saveRequiredFieldsAction(Request $request): RedirectResponse
+    {
+        $addressRequiredFieldsForm = $this->getRequiredFieldsForm();
+        $addressRequiredFieldsForm->handleRequest($request);
+
+        if ($addressRequiredFieldsForm->isSubmitted() && $addressRequiredFieldsForm->isValid()) {
+            $data = $addressRequiredFieldsForm->getData();
+
+            try {
+                $this->getCommandBus()->handle(new SetRequiredFieldsForAddressCommand($data['required_fields']));
+            } catch (AddressException $e) {
+                $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages($e)));
+
+                return $this->redirectToRoute('admin_addresses_index');
+            }
+
+            $this->addFlash('success', $this->trans('Update successful', 'Admin.Notifications.Success'));
```

Comment:
You could move this statement to `try` block and remove `return` from `catch`, thus optimizing code a little. :)